src/main/java/org/broadinstitute/hellbender/utils/variant/writers/GVCFBlockCombiner.java

```diff
@@ -0,0 +1,220 @@
+package org.broadinstitute.hellbender.utils.variant.writers;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.collect.Range;
+import com.google.common.collect.RangeMap;
+import com.google.common.collect.TreeRangeMap;
+import htsjdk.variant.variantcontext.Allele;
+import htsjdk.variant.variantcontext.Genotype;
+import htsjdk.variant.variantcontext.VariantContext;
+import htsjdk.variant.vcf.VCFConstants;
+import htsjdk.variant.vcf.VCFHeader;
+import htsjdk.variant.vcf.VCFHeaderLine;
+import htsjdk.variant.vcf.VCFStandardHeaderLines;
+import org.broadinstitute.hellbender.exceptions.GATKException;
+import org.broadinstitute.hellbender.utils.Utils;
+import org.broadinstitute.hellbender.utils.iterators.PushPullTransformer;
+import org.broadinstitute.hellbender.utils.variant.GATKVCFConstants;
+import org.broadinstitute.hellbender.utils.variant.GATKVCFHeaderLines;
+
+import java.util.ArrayDeque;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Queue;
+
+import static htsjdk.variant.vcf.VCFConstants.MAX_GENOTYPE_QUAL;
+import static org.broadinstitute.hellbender.utils.variant.writers.GVCFWriter.GVCF_BLOCK;
+
+public final class GVCFBlockCombiner implements PushPullTransformer<VariantContext> {
```

Comment:
Laura made changes to the gvcf writer which I think we'll have to update here.. since I wrote this way before then.